### DISCUSSION TITLE
How to RESET react-number-format using React Hook Form

### DISCUSSION THREAD

**Discussion by onlyhasbi** [react-hook-form/react-hook-form#10265] [Q&A]
check example here [reset](https://codesandbox.io/s/reset-react-number-pattern-0q3q9z?file=/src/App.tsx)

**Comment by Sajarin-M** [comment]
Set `defaultValues` for the form

```js
useForm({
  defaultValues: {
   username: "",
   number: ""
  }
});
```

**Reply by onlyhasbi** [reply]
not work

**Reply by onlyhasbi** [reply]
i misplaced onChange handler to onValueChange in react number format, it should be onChange={onChange}

**Reply by Sajarin-M** [reply]
When you use `onChange` instead of `onValueChange`, the value is stored in your formState as formatted.
For example, if you have a form field for a number with the value `455 555`, using `onChange` will store the value in the formState as `455 555`, preserving the spaces between the digits.
